smoketest/src/helper.rs

```diff
@@ -1,31 +1,34 @@
 use crate::constants::*;
 use crate::contracts::i_gateway;
+use crate::contracts::i_gateway::IGateway;
 use crate::parachains::bridgehub::{self};
 use crate::parachains::template::api::runtime_types::xcm as templateXcm;
 use crate::parachains::template::{self};
+use codec::Encode;
 use ethers::prelude::{
     Address, EthEvent, LocalWallet, Middleware, Provider, Signer, SignerMiddleware,
     TransactionRequest, Ws, U256,
 };
 use ethers::providers::Http;
 use futures::StreamExt;
-use sp_core::{sr25519::Pair, Pair as PairT, H160};
+use sp_core::{blake2_256, sr25519::Pair, Pair as PairT, H160};
 use std::ops::Deref;
 use std::sync::Arc;
 use std::time::Duration;
 use subxt::blocks::ExtrinsicEvents;
 use subxt::events::StaticEvent;
 use subxt::tx::{PairSigner, TxPayload};
+use subxt::utils::AccountId32;
 use subxt::{Config, OnlineClient, PolkadotConfig};
 use templateXcm::{
     v3::{junction::Junction, junctions::Junctions, multilocation::MultiLocation},
     VersionedMultiLocation, VersionedXcm,
 };
 
 /// Custom config that works with Statemint
-pub enum TemplateConfig {}
+pub enum AssetHubConfig {}
```

Comment:
Why are you using AssetHub here instead of the template parachain?

Reply:
Actually for template parachain the default `PolkadotConfig` should be fine without a custom `RuntimeConfig`, there is some nuance [here](https://github.com/paritytech/cumulus/blob/742d68753ceb8e9aa5ccfc5993800349a7260b8a/parachains/runtimes/assets/asset-hub-kusama/src/lib.rs#L886) for `AssetHub` which requires it.

Now we don't have smoke tests to cover the asset register/transfer flow from substrate to ethereum direction yet, we'll need it then I think.